src/unix/posix.h

```diff
@@ -62,7 +62,15 @@ GIT_INLINE(int) p_fsync(int fd)
 #define p_snprintf snprintf
 #define p_mkstemp(p) mkstemp(p)
 #define p_chdir(p) chdir(p)
+#ifndef __ANDROID__
 #define p_chmod(p,m) chmod(p, m)
+#else
+/* Pre-Android 5 did not implement a virtual filesystem atop FAT partitions for
```

Comment:
Would it make sense to also gate this depending on the value of `__ANDROID_API__`? That way on the Android versions where this does matter uses the real `chmod(2)`?

Reply:
The problem occurs at runtime so I guess it could work to not use the no-op if a sufficiently new Android platform is the build target. Doing `statfs` on every `chmod` call seems excessive when the `chmod` doesn't ultimately have an effect anyway.